conda_smithy/templates/travis.yml.tmpl

```diff
@@ -4,7 +4,6 @@
 language: generic
 
 os: osx
-osx_image: xcode6.4
```

Comment:
Dropping this entirely is not a good idea. We may consider changing it and installing an old SDK, but dropping it entirely removes a level of control that we need.

Reply:
The new compiler allow us control over that. This is the way we want to go.

Again, this PR is to start discussing that.